File: php/src/TripServiceKata/Trip/TripService.php
<?php

namespace TripServiceKata\Trip;

use TripServiceKata\User\User;
use TripServiceKata\User\UserSession;
use TripServiceKata\Exception\UserNotLoggedInException;

class TripService
{
    public function getTripsByUser(User $user) {
        $tripList = array();
        $loggedUser = $this->getLoggedUser();
        $isFriend = false;
        if ($loggedUser != null) {
            foreach ($user->getFriends() as $friend) {
                if ($friend == $loggedUser) {
                    $isFriend = true;
                    break;
                }
            }
            if ($isFriend) {
                $tripList = $this->findTripsFor($user);
            }
            return $tripList;
        } else {
            throw new UserNotLoggedInException();
        }
    }

    public function getLoggedUser() {
        return UserSession::getInstance()->getLoggedUser();
    }

    public function findTripsFor($user) {
        return TripDAO::findTripsByUser($user);
    }

}

File: java/trip-service-kata/src/main/java/org/craftedsw/tripservicekata/trip/TripService.java
package org.craftedsw.tripservicekata.trip;

import java.util.ArrayList;
import java.util.List;

import org.craftedsw.tripservicekata.user.UserNotLoggedInException;
import org.craftedsw.tripservicekata.user.User;


public class TripService {

	TripDAO tripDAO = null;

	public TripService() {
	}

	public TripService(TripDAO tripDAO) {
		this.tripDAO = tripDAO;
	}

	public List<Trip> getTripsByUser(User user, User loggedUser) throws UserNotLoggedInException {
		if (loggedUser == null) {
			throw new UserNotLoggedInException();
		}

		return user.isFriend(loggedUser) ? findTripsByUser(user) : new ArrayList<>();
	}

	public List<Trip> findTripsByUser(User user) {
		return tripDAO.findTripsByUser(user);
	}

}
